tempest-dynamodb-local/build.gradle.kts

```diff
@@ -65,9 +70,35 @@ tasks.shadowJar {
 
 configure<MavenPublishBaseExtension> {
   configure(
-    JavaLibrary(
-      javadocJar = JavadocJar.Empty(),
-      sourcesJar = false,
-    )
+    KotlinJvm(javadocJar = Dokka("dokkaGfm"))
   )
+
+  pom {
+    withXml {
+      val root = asNode()
+
+      // First collect all dependencies nodes.
+      val dependenciesNodes = root.children()
+        .filterIsInstance<groovy.util.Node>()
+        .filter { it.name().toString().contains("dependencies") }
+        .toList()
+
+      // Then remove them safely.
+      dependenciesNodes.forEach { node ->
+        root.remove(node)
+      }
+
+      // Add a new dependencies node with shadow configuration.
+      val dependenciesNode = root.appendNode("dependencies")
+
+      // Add all shadow dependencies to the POM.
```

Comment:
The net effect here is to remove the dependencies that were shaded, right? Kinda surprised this needs to be done explicitly but I don't have much experience with the intersection of shadow and publication

Reply:
Yeah, this surprised me as well. What's happening is that the publication is using the runtime classpath, which is not the same as the shadow classpath of the shadow jar. IF you publish via the shadow plugin, it can handle this for you, but I didn't want to change the publication mechanism.

Reply:
I didn't want to switch which plugin was used to publish to maven, but if we're okay with doing this via the shadow plugin and it's easier then let's do it.